pkg/volume/gcepd/attacher.go

```diff
@@ -138,6 +141,52 @@ func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(specs []*volume.Sp
 	return volumesAttachedCheck, nil
 }
 
+
+// search Windows disk number by LUN
+func getDiskID(pdName string, exec mount.Exec) (string, error) {
+	cmd := `Get-GcePdName | select Name, DeviceId | ConvertTo-Json`
+	output, err := exec.Run("powershell", "/c", cmd)
+	if err != nil {
+		klog.Errorf("Get-GcePdName failed, error: %v, output: %q", err, string(output))
+		err = errors.New(err.Error() + " " + string(output))
+		return "", err
+	}
+
+	var data []map[string]interface{}
+	if err = json.Unmarshal(output, &data); err != nil {
+		klog.Errorf("Get-Disk output is not a json array, output: %q", string(output))
+		return "", err
+	}
+
+	for _, pd := range data {
+		if jsonName, ok := pd["Name"]; ok {
+			if name, ok := jsonName.(string); ok {
+				klog.Infof("data name %q", name)
```

Comment:
Remove this log line if it's mainly for debugging

Reply:
fixed